packages/devtools/src/commands/performActions.js

```diff
@@ -0,0 +1,109 @@
+import USKeyboardLayout from 'puppeteer-core/lib/USKeyboardLayout'
+
+const sleep = (time = 0) => new Promise(
+    (resolve) => setTimeout(resolve, time))
+
+export default async function performActions ({ actions }) {
+    const page = this.getPageHandle()
+    const lastPointer = {}
+
+    /**
+     * see https://github.com/jlipps/simple-wd-spec#input-sources-and-corresponding-actions
+     * for details on the `actions` format
+     */
+    for (const action of actions) {
+        if (action.type === null || action.type === 'null') {
+            for (const singleAction of action.actions) {
+                await sleep(singleAction.duration)
+            }
+            continue
+        }
+
+        if (action.type === 'key') {
+            const skipChars = []
+            for (const singleAction of action.actions) {
+                if (singleAction.type === 'pause') {
+                    await sleep(singleAction.duration)
+                    continue
+                }
+
+                const cmd = singleAction.type.slice(3).toLowerCase()
```

Comment:
What if we avoid using constants like 3 or 7 and use action.type length instead or something like that to make it more understandable? Maybe define function to get cmd based on action.type

Same for pointer